accepted/00000-enable-ansible-control-node-in-registered-system.md

```diff
@@ -0,0 +1,136 @@
+- Show the systems that are part of this Ansible control node inventory - grouped by onboarding status
+- Bootstrap a selection (from None to All) of the Ansible managed clients
+- A list of the available playbooks on this Ansible control node under a given path - this could be also be another page/subtab.
+- Ultimately, allow to trigger the execution of those playbooks via the UI
+
+#### How to identify if a gathered Ansible managed system is already registered in Uyuni?
+
+- At the time of collecting the Ansible inventory, we only know about the "fqdn" of the Ansible managed client, so only chance at this point if trying to match the new client with an already existing client with the same fqdn.
+
+#### What happen if a gathered Ansible managed system is already registered as traditional client?
+
+- TBD: In this case we do match the system and allow to trigger the bootstrap/migration to a Salt minion in an easy way.
+
+
+### Easy bootstrap of Ansible managed clients that are not yet registered as Salt minion
+
+After collecting the inventory, and matching the existing registered systems, there are probably some Ansible managed clients that are not yet even registered in Uyuni at all. For these systems, via the "Ansible" tab of the registered Ansible control node, we allow to bootstrap a selection of them as Salt minion (or SSH minion), from None to All.
+
+At this point, Uyuni doesn't know how to reach those systems, we only know their "fqdns" according to the Ansible inventory, but the Ansible inventory in the control node should contain information about how to reach them via ssh. The idea here is the following:
+
+In order to bootstrap and trigger the registration of these systems, we have two alternatives:
+
+#### Use "salt-ssh" from Uyuni directly to the "fqdns" of the Ansible managed clients:
+
+* We previously need to fetch the corresponding SSH key that is used to connect to this particular host by Ansible.
+* Then we would reuse this key as SSH credentials to trigger the bootstrap on that host.
+* Downside: Uyuni needs to have direct access to the Ansible managed clients to bootstrap
+
+#### Use "salt-ssh" from Uyuni using the Ansible control node as SSH proxy:
+
+* We do not necessary need to fetch the corresponding SSH key, but we need to have the Uyuni SaltSSH deployed on the Ansible control node to access it
+* The SSH key to connect the Ansible managed client referenced in the Ansible inventory is used to build the ProxyCommand to use in "salt-ssh".
+
+NOTE: It needs to be clarified in the documentation that `ansible_ssh_private_key_file` needs to be defined in the Ansible inventory of your Ansible control node.
+
+* Once we have this, we can trigger the bootstrap of those systems by calling "salt-ssh" like this:
```

Comment:
We will be using salt-ssh but clients may be salt-minions or salt-ssh clients, right?

Reply:
That's my understanding, yes. Just like regular boostrapping which is executed over salt-ssh for both salt-ssh and salt-minion clients

Reply:
Yes, that's correct.